resources/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/CharsetManager.java

```diff
@@ -421,46 +440,74 @@ public void projectPreferencesChanged(IProject project) {
 		charsetListener.charsetPreferencesChanged(project);
 	}
 
-	public void setCharsetFor(IPath resourcePath, String newCharset) throws CoreException {
-		// for the workspace root we just set a preference in the instance scope
-		if (resourcePath.segmentCount() == 0) {
-			IEclipsePreferences resourcesPreferences = InstanceScope.INSTANCE.getNode(ResourcesPlugin.PI_RESOURCES);
-			if (newCharset != null)
-				resourcesPreferences.put(ResourcesPlugin.PREF_ENCODING, newCharset);
-			else
-				resourcesPreferences.remove(ResourcesPlugin.PREF_ENCODING);
-			try {
-				resourcesPreferences.flush();
-			} catch (BackingStoreException e) {
-				IProject project = workspace.getRoot().getProject(resourcePath.segment(0));
```

Comment:
This was actually an ArrayIndexOutOfBoundsException

Reply:
"Was" as in "in the old code"? Does the exception still persist?

Reply:
Is the right procedure for me to open an issue?